src/routes/clinical/arranger.ts

```diff
@@ -0,0 +1,95 @@
+import { ADVERTISED_HOST } from 'config';
+import { get } from 'lodash';
+import fetch from 'node-fetch';
+import urljoin from 'url-join';
+
+const url = urljoin(ADVERTISED_HOST, '/graphql');
+
+const hitsQuery = `
+  query hitsQuery($filters: JSON, $first: Int) {
+    file {
+      hits(filters: $filters, first: $first) {
+        edges {
+          node {
+            donors {
+              hits {
+                edges {
+                  node {
+                    donor_id
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }`;
+
+const totalQuery = `
+  query totalQuery($filters: JSON) {
+    file {
+      hits(filters: $filters) {
+        total
+      }
+    }
+  }`;
+
+function makeRequest({ query, variables }: { query: string; variables: Record<string, unknown> }) {
+	return JSON.stringify({
+		query,
+		variables,
+	});
+}
+
+function createQueryClient({ requestConfig }: { requestConfig: Record<string, unknown> }) {
+	return async ({ query }: { query: string }) => {
+		try {
+			const response = await fetch(url, {
+				...requestConfig,
+				body: query,
+			});
+			const data = await response.json();
+			return data;
+		} catch (error) {
+			console.error(`Failed to query server`, error);
+		}
+	};
+}
+
+async function queryArranger({
+	requestFilter,
+	egoJwt,
+}: {
```

Comment:
Powerful new exported function, should have TSDoc describing what it does, what inputs it requires and what it will return. Best to also include example of how to use it.